### PR TITLE
[PAP][Admin] Add campaign history admin

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Admin/Pap/CampaignHistoryAdmin.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Admin\Pap;
+
+use App\Pap\CampaignHistoryStatusEnum;
+use App\Pap\CampaignHistoryVoterStatusEnum;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
```
Comment (ottaviano) [inline src/Admin/Pap/CampaignHistoryAdmin.php:6]:
```suggestion
use App\Admin\AbstractAdmin;
```

File: src/Admin/Pap/CampaignHistoryAdmin.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    protected function configureListFields(ListMapper $listMapper)
+    {
+        $listMapper
+            ->add('campaign', null, [
+                'label' => 'Campagne',
+            ])
+            ->add('building.address', null, [
+                'label' => 'Adresse',
+                'template' => 'admin/pap/campaign_history/list_address.html.twig',
```
Comment (ottaviano) [inline src/Admin/Pap/CampaignHistoryAdmin.php:74]: what about using the existing template with a little helper:
```php
CampaignHistory::getAddress() {
  return $this->building->getAddress();
}
``` 
```suggestion
                'template' => 'admin/list_address.html.twig',
```

File: templates/admin/pap/campaign_history/list_survey.html.twig [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{% extends '@SonataAdmin/CRUD/base_list_field.html.twig' %}
+
+{% block field %}
+    {% if object.dataSurvey %}
+        {% set type = object.dataSurvey.survey is instanceof('App\\Entity\\Jecoute\\NationalSurvey') ? 'national' : 'local' %}
+        <a href="{{ path('admin_app_jecoute_'~type~'survey_edit', { id: object.dataSurvey.survey.id }) }}">
+            {{ object.dataSurvey.survey.name }}
+        </a>
```
Comment (ottaviano) [inline templates/admin/pap/campaign_history/list_survey.html.twig:8]:
```suggestion
        <a href="{{ path('admin_app_jecoute_'~(object.dataSurvey.survey.getType())~'survey_edit', { id: object.dataSurvey.survey.id }) }}">
            {{ object.dataSurvey.survey.name }}
        </a>
```

**Review by ottaviano — APPROVED** [review]

File: src/Admin/Pap/CampaignHistoryAdmin.php (hunk 2 above)
Comment (illusionOfParadise) [reply src/Admin/Pap/CampaignHistoryAdmin.php:74]: This template is for another entity `Address`, it does not match the fields of `PapAddress`.

**Pull request merged**